src/components/Form.js

```diff
@@ -300,6 +300,7 @@ export class Form extends React.Component {
   onSubmit() {
     const {validate} = this.props
     const {source, options} = this.state
+    if (source.endsWith('datapackage.json')) options.preset = 'datapackage'
```

Comment:
That's just an example change to make it support data packages. Should be re-written.